### PR TITLE
Add narrative docs for threadsafe instances.

### PR DESCRIPTION
**Pull request by rfk**
I wrote these docs are part of working through the proposal in https://github.com/mozilla/uniffi-rs/pull/368 and figured I may as well push them for visibility. They may or may not turn out to be a useful starting point for the implementation work connected to #370. /cc @jhugman

### CODE REVIEW

**Comment by rfk** [comment]
(N.B. this PR currently targets #367 rather than `main`, since it builds on top of the docs added therein)

File: docs/manual/src/internals/lifting_and_lowering.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+```idl
+namespace example {
+  sequence<i32> add_to_list(i32 item);
+}
+```
+
+Calling this function from foreign language code involves the following steps:
+
+1. The user-provided calling code invokes the `add_to_list` function that is expoed by the
```
Comment (mhammond) [inline docs/manual/src/internals/lifting_and_lowering.md:31]: nit: exposed

**Review by mhammond — APPROVED** [review]
I didn't get to the end up this, but found a nit I thought I'd share before I take off for the weekend :)

**Comment by rfk** [comment]
This was lightly updated to reflect that actually merged, but I'm going to land it on the strength of Mark's r+.

**Pull request merged**